scalastyle-config.xml

```diff
@@ -264,6 +264,19 @@ This file is divided into 3 sections:
     of Commons Lang 2 (package org.apache.commons.lang.*)</customMessage>
   </check>
 
+  <check customId="FileSystemGet" level="error" class="org.scalastyle.file.RegexChecker" enabled="true">
+    <parameters><parameter name="regex">FileSystem.get\([a-zA-Z_$][a-zA-Z_$0-9]*\)</parameter></parameters>
+    <customMessage><![CDATA[
+      Are you sure that you want to use "FileSystem.get(Configuration conf)"? If the input
+      configuration is not set properly, a default FileSystem instance will be returned. Please use
+      "FileSystem.get(URI uri, Configuration conf)" or "Path.getFileSystem(Configuration conf)" instead.
```

Comment:
Using `FileSystem.get(Configuration conf)` is dangerous even outside of the misconfiguration scenario you describe. It's entirely possible for Spark applications to be working with multiple `FileSystem` instances (say, local HDFS and a cloud blob store, or two HDFS clusters). Using `FileSystem.get(Configuration conf)` in this instance can also result in nasty errors.

Not sure if it's worth mentioning here, will leave that up to your judgement.

Reply:
@xkrogen Thanks for the +1! I just update the message in style check. I think mentioning "reading a target path of non-default file system" is good enough.

Reply:
how about
```
Are you sure that you want to use "FileSystem.get(Configuration conf)"? If the input
configuration is not set properly, a default FileSystem instance will be returned. It can
also lead to nasty errors when you deal with multiple file systems. Please use ....
```

Reply:
@cloud-fan ok, updated.